### ISSUE TITLE
Understanding the deferList size/status from an action

### ISSUE THREAD

**Issue by ottawanempire** [spring-projects/spring-statemachine#348]
Hi, following enhancement would be appreciated:

- There can be conditions where business logic suppose to understand if there is any deferred event waiting to be processed before making a decision in an action.
- Currently, it seems that the deferList within DefaultStateMachineExecutor does not have a public method to expose that information.
- Having a method on StateMachine or Context object to provide the information of deferList would be great. A boolean value to be returned should be enough: false if size equals 0, true if size is greater then 0.

**Comment by jvalkeal** [comment]
Yes, I've also felt time to time that UML spec for deferred events is a bit awkward and relatively loose. Although I have never had a use case to use or know current deferred events, there's been cases that it would have been nice to be able to clear those events.

Maybe we can try to overhaul features around deferred events to be more usable in real world.

**Comment by jingweiz2017** [comment]
Hello, 

My use case is kinda related to this discussion. The use case requires me to reset the statemachine. However, with the current resetstatemachine function, I wouldn't be able to clean those delay events as long as I choose to keep the original statemachine instance in memory. Would it be better to have deferlist on statemachinecontext rather than defaultStatemachineExecutor? So user can choose to reset those event, as well? And it would work for statemachine persist, too. 

I also have a use case in which an event get triggered when the statemachine is on certain state. However, I don't want to trigger that event if we have certain defer events in the queue. I know I can work around it with extendedstatevariables. But, it would be better if I could just check out "What appointments do I have for now" and decide what to do next. 

What do you think?
